WordPressKit/Jetpack Scan/JetpackScanThreat.swift

```diff
@@ -0,0 +1,263 @@
+
+        // Define a placeholder type to prevent the used before initialized error
+        type = .unknown
+
+        id = try container.decode(Int.self, forKey: .id)
+        signature = try container.decode(String.self, forKey: .signature)
+        fileName = try container.decodeIfPresent(String.self, forKey: .fileName)
+        description = try container.decode(String.self, forKey: .description)
+        firstDetected = try container.decode(ISO8601WithMSDate.self, forKey: .firstDetected)
+        fixedOn = try container.decodeIfPresent(ISO8601WithMSDate.self, forKey: .fixedOn)
+        fixable = try? container.decodeIfPresent(JetpackScanThreatFixer.self, forKey: .fixable) ?? nil
+        `extension` = try container.decodeIfPresent(JetpackThreatExtension.self, forKey: .extension)
+        diff = try container.decodeIfPresent(String.self, forKey: .diff)
+        rows = try container.decodeIfPresent([String: Any].self, forKey: .rows)
+        status = (try? container.decode(ThreatStatus.self, forKey: .status)) ?? .unknown
+
+        // Context obj can either be:
+        // - not present
+        // - a dictionary
+        // - an empty string
+        // we can not just set to nil because the threat type logic needs to know if the
+        // context attr was present or not
+        if let contextDict = try container.decodeIfPresent([String: Any].self, forKey: .context) {
+            context = JetpackThreatContext(with: contextDict)
+        } else if ((try container.decodeIfPresent(String.self, forKey: .context)) != nil) {
+            context = JetpackThreatContext.emptyObject()
+        } else {
+            context = nil
+        }
+
+        // Calculate the type of threat last
+        type = ThreatType(threat: self)
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case fileName = "filename"
+        case firstDetected = "first_detected"
+        case fixedOn = "fixed_on"
+        
+        case id, signature, description, fixable
+        case `extension`, diff, context, rows, status
+    }
+}
+
+/// An object that describes how a threat can be fixed
+public struct JetpackScanThreatFixer: Decodable {
+    public enum ThreatFixType: String, Decodable {
+        case replace
+        case delete
+        case update
+        case edit
+
+        case unknown
+    }
+
+    /// The suggested threat fix type
+    public let type: ThreatFixType
+
+    /// The file path of the file to be fixed
+    public var file: String?
+
+    /// The target version to fix to
+    public var target: String?
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        type = (try? container.decode(ThreatFixType.self, forKey: .type)) ?? .unknown
+        file = try container.decodeIfPresent(String.self, forKey: .file)
+        target = try container.decodeIfPresent(String.self, forKey: .target)
+    }
```

Comment:
```suggestion
```

Reply:
From my understanding this wouldn't work because I need the `type` to default to `unknown` if the value of `type` is outside the enum values. 

That's why I left the boilerplate code in that I did, because I need to manually set some of the properties up.

Thoughts?

Reply:
@emilylaguna , have you read this: https://www.latenightswift.com/2019/02/04/unknown-enum-cases/ or similar? Ialso wonder if we can reduce some of the decoder boilerplate code in this PR

Reply:
@yaelirub awesome, I have not seen that. I've gone ahead and implemented that in.